### PR TITLE
Vnet peering test updates

### PR DESCRIPTION
**Pull request by kargakis**
/cc @kwoodson

### CODE REVIEW

**Comment by kargakis** [comment]
Got a successful run
```
Peer Vnet tests [Vnet][Real][LongRunning] 
  should create the vnet and cluster and verify peering
  /home/kargakis/go/src/github.com/openshift/openshift-azure/test/e2e/specs/realrp/vnetpeering.go:57
configuring the real resource provider
time="2018-12-19T14:38:17+01:00" level=info msg="using generated resource group name generated-pg6igahj"
STEP: creating resource group generated-pg6igahj
STEP: creating a custom vnet
time="2018-12-19T14:38:29+01:00" level=debug msg="generating manifest from ../../test/manifests/normal/create.yaml"
STEP: creating an OSA cluster
STEP: ensuring the OSA cluster vnet is peered with the custom vnet
STEP: deleting resource group generated-pg6igahj

• [SLOW TEST:553.516 seconds]
Peer Vnet tests [Vnet][Real][LongRunning]
/home/kargakis/go/src/github.com/openshift/openshift-azure/test/e2e/specs/realrp/vnetpeering.go:20
  should create the vnet and cluster and verify peering
  /home/kargakis/go/src/github.com/openshift/openshift-azure/test/e2e/specs/realrp/vnetpeering.go:57
------------------------------
SSSSSSSSSSSSSSSSSS
Ran 1 of 30 Specs in 553.517 seconds
SUCCESS! -- 1 Passed | 0 Failed | 0 Pending | 29 Skipped
--- PASS: TestE2E (553.52s)
PASS
ok  	github.com/openshift/openshift-azure/test/e2e	553.534s
```

**Comment by kargakis** [comment]
@kwoodson this PR should fix the current issue with the vnet test: https://openshift-gce-devel.appspot.com/build/origin-ci-test/logs/periodic-ci-azure-vnet-e2e/2

File: test/e2e/specs/realrp/vnetpeering.go [hunk 1]
```diff
@@ -78,44 +75,42 @@ var _ = Describe("Peer Vnet tests [Vnet][Real][LongRunning]", func() {
 					},
 				},
 			},
-			Location: &region,
+			Location: &cfg.Region,
 		})
 		Expect(err).ToNot(HaveOccurred())
-		Expect(future).ToNot(BeNil())
```
Comment (kwoodson) [inline test/e2e/specs/realrp/vnetpeering.go]: @charlesakalugwu has asked me to check for nils on the return objects.

Comment (kargakis) [reply test/e2e/specs/realrp/vnetpeering.go]: Futures are [structs](https://github.com/Azure/azure-sdk-for-go/blob/07f918ba2d513bbc5b75bc4caac845e10f27449e/services/devtestlabs/mgmt/2016-05-15/dtl/models.go#L8513) and they are always complete when err == nil

**Pull request merged**